ledger/src/main/java/com/scalar/dl/ledger/database/AssetLedger.java

```diff
@@ -27,6 +29,10 @@ public Optional<com.scalar.dl.ledger.asset.Asset> get(String assetId) {
 
   @Override
   public List<com.scalar.dl.ledger.asset.Asset> scan(AssetFilter filter) {
+    if (filter instanceof NamespaceAwareAssetFilter) {
+      throw new ContractException(
+          CommonError.NAMESPACE_AWARE_ASSET_FILTER_UNSUPPORTED_IN_DEPRECATED_CONTRACT);
+    }
```

Comment:
FYI: I didn't add namespace-aware interfaces to the deprecated Ledger, preventing their use in the old `Contract` class, since we shouldn't encourage using them.